src/RegExLib.Core/Entities/Expression.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System;
+using RegExLib.Core.Events;
+using RegExLib.Core.Interfaces;
+using RegExLib.SharedKernel;
+
+namespace RegExLib.Core.Entities
+{
+    public class Expression : BaseEntity
+    {
+        public int AuthorId { get; set; }
+        public string Title { get; set; }
+        public string Pattern { get; set; }
+        public string Description { get; set; }
+        public Author Author { get; set; }
```

Comment:
remove setters for required properties?

Reply:
I changed it to this

```csharp
        public readonly int AuthorId;
        public readonly string Title;
        public readonly string Pattern;
        public readonly string Description;
        public readonly Author Author;

        public Expression(string title, string pattern, string description, Author author)
        {
            Title = title;
            Pattern = pattern;
            Description = description;
            Author = author;
            AuthorId = author.Id;
        }
```

Reply:
You need private setters and (at least a ) private parameterless constructor in order for EF to be able to fetch instances from persistence.

Reply:
@ardalis 

> You need private setters and (at least a ) private parameterless constructor in order for EF to be able to fetch instances from persistence.

When I first read this, I was 😧! But I was rewatching you and Julie’s Pluralsight course last night and came upon this:

![image](https://user-images.githubusercontent.com/17127310/86121312-d093bc80-baa3-11ea-8e14-d32e293ba153.png)

😀

Reply:
:)